### PR TITLE
Add a EnrolmentId control

### LINKED ISSUE #2170: Update HTC encounter
- [x] For TB and GBV screening, follow HPDB instead of the paper based forms
- [x] Update HTC dashboard report using new TB screening fields
- [x] Add enrolmentId control to display id 
- [x] Fix Patient name in encounter if patient doesn't has a first and last name (yet)

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #2170

Add missing control for the HIV testing encounter. The control shows the enrolment ID of a certain enrolment type.

This control will also be needed for contact tracing entries. However, it will some tweaking to get the contact's patient id then...

# Testing
Create a HIV testing encounter adn got to: "Hiv Testing" and select Result "Positive". The ART Number should show up (but blank).

Enrol into a the HIV care encounter and assign a enrolment id during enrolment. Going back to the testing encounter, the ART Number should now show a value.

image

### CODE REVIEW

File: client/packages/programs/src/api/api.ts [hunk 1]
```diff
@@ -258,7 +259,7 @@ export const getAllocateProgramNumber = (sdk: Sdk, storeId: string) => ({
 
 export type ProgramEnrolmentListParams = {
   sortBy?: SortRule<ProgramEnrolmentSortFieldInput>;
-  filterBy?: FilterBy;
+  filterBy?: ProgramEnrolmentFilterInput;
```
Comment (clemens-msupply) [inline client/packages/programs/src/api/api.ts:262]: Using this type from the generated API gives the correct autocompletion, type checks, etc...

File: client/packages/programs/src/JsonForms/components/EnrolmentId.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+const UIComponent = (props: ControlProps) => {
+  const { label, uischema, config } = props;
+
+  const { errors, options } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+
+  // fetch current encounter
```
Comment (CarlosNZ) [inline client/packages/programs/src/JsonForms/components/EnrolmentId.tsx:28]: Need to update comment here I think -- "fetch enrolments"?

Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/components/EnrolmentId.tsx:28]: Thanks done!

File: client/packages/programs/src/JsonForms/components/EnrolmentId.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          ? enrolment.programEnrolmentId
+          : '',
+        disabled: true,
+        required: props.required,
+        error: !!errors,
+        helperText: errors,
+        sx: { ...inputProps.sx, ...DefaultFormRowSpacing },
+      }}
+      labelWidthPercentage={FORM_LABEL_WIDTH}
+      inputAlignment={'start'}
```
Comment (mark-prins) [inline client/packages/programs/src/JsonForms/components/EnrolmentId.tsx:63]: it's only a string 🤷 
```suggestion
      inputAlignment="start"
```

**Review by mark-prins — APPROVED** [review]
👍  woop! it's satisfying when things work as advertised

**Pull request merged**